src/vs/workbench/contrib/debug/browser/debugSession.ts

```diff
@@ -905,6 +905,28 @@ export class DebugSession implements IDebugSession {
 
 		const outputQueue = new Queue<void>();
 		this.rawListeners.push(this.raw.onDidOutput(async event => {
+			// When a variables event is received, execute immediately to obtain the variables value #126967
+			if (event.body.variablesReference) {
+				const source = event.body.source && event.body.line ? {
+					lineNumber: event.body.line,
+					column: event.body.column ? event.body.column : 1,
+					source: this.getSource(event.body.source)
+				} : undefined;
+				const container = new ExpressionContainer(this, undefined, event.body.variablesReference, generateUuid());
+				await container.getChildren().then(children => {
+					children.forEach(child => {
+						// Since we can not display multiple trees in a row, we are displaying these variables one after the other (ignoring their names)
+						(<any>child).name = null;
+						// we should put appendToRepl into queue to make sure the logs to be displayed in correct order
+						// see https://github.com/microsoft/vscode/issues/126967#issuecomment-874954269
+						outputQueue.queue(async () => {
```

Comment:
The enqueuing of output must happen synchronously to preserve order. The `onDidOutput` listener should not be async at all.

What you can do is something like

```
const children = container.getChildren();
outputQueue.queue(async () => {
    const resolved = await children;
    resolved.forEach(() => ...)
});
```

Reply:
done.